### DISCUSSION TITLE
[FE210] 全域變數問題

### DISCUSSION THREAD

**Discussion by rdcvgt** [Lidemy/forum#285] [其他問題] [answered]
這邊想問個簡單問題，但我目前還沒有什麼頭緒

因為從單元中的 infinite scroll 到 i18n，看到 Huli 用` window.isLoading`,` window.I18N` 的方式來宣告全域變數
這邊我有點好奇  Huli 選擇用 window.variable 而不是 var, let （或 const）的方式來宣告的原因是什麼呢？（或是說通常什麼狀況下會決定使用  window.variable ）

我有上網查到[相關討論](https://stackoverflow.com/questions/6349232/whats-the-difference-between-a-global-variable-and-a-window-variable-in-javas)，但不確定說在這堂課如此使用的原因為何，再麻煩老師解答了！

（我原本還把 window.I18N 跟 window.scrollY 的觀念搞混，以為都是 JS 內建用法，結果後來看 Huli 的說明才發現 window.I18N 是自訂的全域變數 XDD）

**Comment by aszx87410** [comment, marked as answer]
用 `window.I18N` 是因為在範例中語言檔是透過額外的 script 載入進去的，如果不用全域變數的話就無法溝通
